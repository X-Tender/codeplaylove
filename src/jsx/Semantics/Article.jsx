import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Article = ({ className, children, isPage }) => {
	const classes = classNames('article', className, { 'article--page-grid': isPage });
	return <article className={classes}>{children}</article>;
};

Article.propTypes = {
	className: PropTypes.string,
	children: PropTypes.node,
	isPage: PropTypes.bool,
};

Article.defaultProps = {
	className: '',
	children: null,
	isPage: false,
};

export default Article;
