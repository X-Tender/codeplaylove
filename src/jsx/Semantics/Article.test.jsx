import React from 'react';
import { shallow } from 'enzyme';
import Article from './Article';

describe('Article', () => {
	const className = 'article';
	const children = <p>Hello there</p>;

	describe('with all props', () => {
		const articleWithAllProps = shallow(
			<Article className={className} isPage>
				{children}
			</Article>
		);

		it('renders correctly', () => {
			expect(articleWithAllProps).toMatchSnapshot();
		});

		it('renders props correctly', () => {
			const articleWithProps = articleWithAllProps.find('article');

			expect(articleWithProps.hasClass(className)).toBe(true);
			expect(articleWithProps.hasClass('article--page-grid')).toBe(true);
			expect(articleWithProps.contains(children)).toBe(true);
		});
	});

	describe('with no props', () => {
		const articleWithAllProps = shallow(<Article />);

		it('renders correctly', () => {
			expect(articleWithAllProps).toMatchSnapshot();
		});

		it('renders props correctly', () => {
			const articleWithoutProps = articleWithAllProps.find('article');

			expect(articleWithoutProps.hasClass('article--page-grid')).toBe(false);
			expect(articleWithoutProps.children()).toHaveLength(0);
		});
	});
});
