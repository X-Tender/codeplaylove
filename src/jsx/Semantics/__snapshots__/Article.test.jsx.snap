// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article with all props renders correctly 1`] = `
<article
  className="article article article--page-grid"
>
  <p>
    Hello there
  </p>
</article>
`;

exports[`Article with no props renders correctly 1`] = `
<article
  className="article"
/>
`;
